pynest/examples/wang_decision_making.py

```diff
@@ -0,0 +1,437 @@
+
+    # neuron parameters
+    epop_params = {
+        "tau_GABA": 5.0,             # GABA decay time constant
+        "tau_AMPA": 2.0,             # AMPA decay time constant
+        "tau_decay_NMDA": 100.0,     # NMDA decay time constant
+        "tau_rise_NMDA": 2.0,        # NMDA rise time constant
+        "alpha": 0.5,                # NMDA parameter
+        "conc_Mg2": 1.0,             # Magnesium concentration
+        "g_L": 25.0,                 # leak conductance
+        "E_L": -70.0,                # leak reversal potential
+        "E_ex": 0.0,                 # excitatory reversal potential
+        "E_in": -70.0,               # inhibitory reversal potential
+        "V_reset": -55.0,            # reset potential
+        "V_th": -50.0,               # threshold
+        "C_m": 500.0,                # membrane capacitance
+        "t_ref": 2.0,                # refreactory period
+    }
+
+    ipop_params = {
+        "tau_GABA": 5.0,             # GABA decay time constant
+        "tau_AMPA": 2.0,             # AMPA decay time constant
+        "tau_decay_NMDA": 100.0,     # NMDA decay time constant
+        "tau_rise_NMDA": 2.0,        # NMDA rise time constant
+        "alpha": 0.5,                # NMDA parameter
+        "conc_Mg2": 1.0,             # Magnesium concentration
+        "g_L": 20.0,                 # leak conductance
+        "E_L": -70.0,                # leak reversal potential
+        "E_ex": 0.0,                 # excitatory reversal potential
+        "E_in": -70.0,               # inhibitory reversal potential
+        "V_reset": -55.0,            # reset potential
+        "V_th": -50.0,               # threshold
+        "C_m": 200.0,                # membrane capacitance
+        "t_ref": 1.0,                # refreactory period
+    }
+    # fmt: on
+
+    # Signals to the two different excitatory sub-populations
+    # the signal is given by a time-inhomogeneous Poisson process,
+    # where the expectations are constant over intervals of 50ms,
+    # and then change. The values for each interval are normally
+    # distributed, with means `mu_a`and `mu_b`, and standard deviation
+    # `sigma`.
+    signal_start = 1000.0
+    signal_duration = 1000.0
+    signal_update_interval = 50.0
+    f = 0.15  # proportion of neurons receiving signal inputs
+    # compute expectations of the time-inhomogeneous Poisson processes
+    mu_0 = 40.0  # base rate
+    rho_a = mu_0 / 100  # scaling factors coherence
+    rho_b = rho_a
+    sigma = 4.0  # standard deviation
+    mu_a = mu_0 + rho_a * coherence  # expectation for pop A
+    mu_b = mu_0 - rho_b * coherence  # expectation for pop B
+
+    # sample values for the Poisson process
+    num_updates = int(signal_duration / signal_update_interval)
+    update_times = np.arange(0, signal_duration, signal_update_interval)
+    update_times[0] = 0.1
+
+    # for p ... below allows us to create a local variable in the iterator which is created once
```

Comment:
This comment looks a bit uncomplete

Reply:
Your are right, here an improved version:
```suggestion
    # We could have written the generator expressions passed to `np.fromiter()` below as
    #
     #    ( nest.CreateParameter(...).GetValue() for _ in range(num_updates) )
     #
     # but that would create a new Parameter object for each iteration. We avoid this by
     # wrapping the iterator in an outer loop in which `p` assumes only a single value.
```